File: category/impl/src/main/java/thirdpower/mydms/category/impl/CategoryModule.java
package thirdpower.mydms.category.impl;

import com.google.auto.service.AutoService;
import com.google.inject.AbstractModule;
import com.google.inject.Module;

import thirdpower.mydms.category.api.CategoryService;

@AutoService(Module.class)
public class CategoryModule extends AbstractModule {

  @Override
  protected void configure() {
    bind(CategoryService.class).to(DefaultCategoryService.class);
  }
}

File: category/impl/src/main/kotlin/thirdpower/mydms/category/impl/CategoryModule.kt
package thirdpower.mydms.category.impl

import com.google.inject.AbstractModule
import com.google.auto.service.AutoService
import com.google.inject.Module
import thirdpower.mydms.category.api.CategoryService

@AutoService(Module::class)
class CategoryModule : AbstractModule() {
	
	override fun configure() {
		bind(CategoryService::class.java).to(DefaultCategoryService::class);
	}
}